src/OpenTelemetry.Collector.AspNet/Implementation/HttpInListener.cs

```diff
@@ -0,0 +1,144 @@
+﻿// <copyright file="HttpInListener.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Web;
+using System.Web.Routing;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Collector.AspNet.Implementation
+{
+    internal class HttpInListener : ListenerHandler
+    {
+        private readonly PropertyFetcher routeFetcher = new PropertyFetcher("Route");
+        private readonly PropertyFetcher routeTemplateFetcher = new PropertyFetcher("RouteTemplate");
+        private readonly AspNetCollectorOptions options;
+
+        public HttpInListener(string name, Tracer tracer, AspNetCollectorOptions options)
+            : base(name, tracer)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStartActivity";
+
+            var context = HttpContext.Current;
+            if (context == null)
+            {
+                CollectorEventSource.Log.NullPayload(nameof(HttpInListener) + EventNameSuffix);
+                return;
+            }
+
+            if (this.options.RequestFilter != null && !this.options.RequestFilter(context))
```

Comment:
RequestFilter can be a user supplied function right? Would it make sense to try catch it and log an Error level message to EventSource, so user can be made aware of the issue.

Reply:
Will be caught and written here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry/Collector/DiagnosticSourceListener.cs#L69

IMO given the rarity, should be fine?